### PR TITLE
add libglvnd-devel

### PR DESCRIPTION
**Pull request by basnijholt**
This is required for https://github.com/conda-forge/paraview-feedstock/pull/126/

Please see the repo readme for directions on how to make PRs on this repo.

Checklist:

- [x] if you have added a CDT, it appears in the `cdt_slugs.yaml` file and 
  you have rerun the script `python gen_cdt_recipes.py`.
- [ ] if you have changed the CDT generator code (`rpm.py`), you have bumped
  the build number in `conda_build_config.yaml` and have remade all of the
  recipes via running `python gen_cdt_recipes.py --force`
- [ ] if you have added a custom CDT recipe, you have added the name of the CDT
  with `custom: true` in the `cdt_slugs.yaml` file.
- [ ] all CDT recipes have build number set by `{{ cdt_build_number }}` for
  old-style/legacy CDTs or `{{ cdt_build_number|int + 1000 }}` for new-style CDTs
- [ ] if you see a warning about a CDT not having a license, you have added the
  `license_file` key in the `cdt_slugs.yaml` file with the path to the appropriate
  license in `licenses/`

**NOTE: If you make any changes to `cd_slugs.yaml`, you need to reun the generator code
via `python gen_cdt_recipes.py`. Add the `--fast` option to make this go more quickly.**

### CODE REVIEW

**Comment by basnijholt** [comment]
@conda-forge/help-cdts, could someone please have a look at this?

I have run `python gen_cdt_recipes.py` what created many changes. Is this supposed to happen?

This is blocking https://github.com/conda-forge/paraview-feedstock/pull/126.

**Comment by beckermr** [comment]
It looks like you may have hit a bug of some sort. It should not have changed the other recipes.

**Comment by basnijholt** [comment]
@beckermr, thanks. I have removed all the other files from the commits. Should I also include the legacy_cdt files for the new `libglvnd-devel` I am adding?

**Comment by beckermr** [comment]
Yeah let's build everything. I'll need to debug the issues in the generation locally. I want to do that before merging to make sure we can regenerate the recipes without trouble.

**Comment by beckermr** [comment]
You shouldn't have had to add the packages to the readme. Something is wrong...

**Comment by basnijholt** [comment]
@beckermr I have switched to a different (Linux) machine and installed a fresh environment using `env.yml`.

Now rerunning the script only gives me additional changes of the following form:
```diff
diff --git a/cdts/ibacm-cos7-aarch64/meta.yaml b/cdts/ibacm-cos7-aarch64/meta.yaml
index fcc83a2..a191c80 100644
--- a/cdts/ibacm-cos7-aarch64/meta.yaml
+++ b/cdts/ibacm-cos7-aarch64/meta.yaml
@@ -3,11 +3,11 @@ package:
   version: 22.4

 source:
-  - url: http://mirror.centos.org/altarch/7/os/aarch64/Packages/ibacm-22.4-1.el7.aarch64.rpm
-    sha256: a7f447298f3fdaa87e3e9896b2720dc0747a7d3361f22e483daa2b789106d3e3
+  - url: http://mirror.centos.org/altarch/7/os/aarch64/Packages/ibacm-22.4-5.el7.aarch64.rpm
+    sha256: ccd063c99b81ed96b11de2b1a159d3d1cafcf6383e9691b9cb81e746de9c265b
     no_hoist: true
     folder: binary
-  # - url: http://vault.centos.org/7.7.1908/os/Source/SPackages/rdma-core-22.4-1.el7.src.rpm
+  # - url: http://vault.centos.org/7.7.1908/os/Source/SPackages/rdma-core-22.4-5.el7.src.rpm
   #   no_hoist: true
   #   folder: source

diff --git a/cdts/ibacm-cos7-ppc64le/meta.yaml b/cdts/ibacm-cos7-ppc64le/meta.yaml
index 122928c..c1abc2e 100644
--- a/cdts/ibacm-cos7-ppc64le/meta.yaml
+++ b/cdts/ibacm-cos7-ppc64le/meta.yaml
@@ -3,11 +3,11 @@ package:
   version: 22.4

 source:
-  - url: http://mirror.centos.org/altarch/7/os/ppc64le/Packages/ibacm-22.4-1.el7.ppc64le.rpm
-    sha256: 4f1228549ca4ada161d59493b5b6a528dcbe3e9360f37039f4181445b7aa72f0
+  - url: http://mirror.centos.org/altarch/7/os/ppc64le/Packages/ibacm-22.4-5.el7.ppc64le.rpm
+    sha256: cd7997797d3f64635e58157924d2e5aa996a02a07421ab24c7ce7adf6bdb9a56
     no_hoist: true
     folder: binary
-  # - url: http://vault.centos.org/7.7.1908/os/Source/SPackages/rdma-core-22.4-1.el7.src.rpm
+  # - url: http://vault.centos.org/7.7.1908/os/Source/SPackages/rdma-core-22.4-5.el7.src.rpm
   #   no_hoist: true
   #   folder: source
```

Which is probably fine?

Also the builds succeed! :tada:

Is there anything else I can do to get this merged?

**Comment by beckermr** [comment]
Ahhh great! So we'll still need to change the scripts to make sure it bumps the build number for old recipes that change. It appears new builds of the upstream RPMs are being inserted. 

Let's back those extra changes out for now by applying the diff above. Then we can merge.

**Comment by basnijholt** [comment]
Sorry, I do not really follow what you are suggesting now.

I have not committed any of the changes that are unrelated to `libglvnd`. And how/where can I change the build numbers?

These are the only files that are changed:
```bash
(cdt-builds) cw-basnijholt  ➜  cdt-builds git:(libglvnd-devel) ✗  gst
 README.md                                                     |  15 +++
 cdt_slugs.yaml                                                |  12 +++
 cdts/libglvnd-core-devel-cos7-aarch64/build.sh                |  59 +++++++++++
 cdts/libglvnd-core-devel-cos7-aarch64/libglvnd-license        | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 cdts/libglvnd-core-devel-cos7-aarch64/meta.yaml               |  46 +++++++++
 cdts/libglvnd-core-devel-cos7-ppc64le/build.sh                |  59 +++++++++++
 cdts/libglvnd-core-devel-cos7-ppc64le/libglvnd-license        | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 cdts/libglvnd-core-devel-cos7-ppc64le/meta.yaml               |  46 +++++++++
 cdts/libglvnd-core-devel-cos7-x86_64/build.sh                 |  59 +++++++++++
 cdts/libglvnd-core-devel-cos7-x86_64/libglvnd-license         | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 cdts/libglvnd-core-devel-cos7-x86_64/meta.yaml                |  46 +++++++++
 cdts/libglvnd-devel-cos7-aarch64/build.sh                     |  66 ++++++++++++
 cdts/libglvnd-devel-cos7-aarch64/libglvnd-license             | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 cdts/libglvnd-devel-cos7-aarch64/meta.yaml                    |  62 ++++++++++++
 cdts/libglvnd-devel-cos7-ppc64le/build.sh                     |  66 ++++++++++++
 cdts/libglvnd-devel-cos7-ppc64le/libglvnd-license             | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 cdts/libglvnd-devel-cos7-ppc64le/meta.yaml                    |  62 ++++++++++++
 cdts/libglvnd-devel-cos7-x86_64/build.sh                      |  66 ++++++++++++
 cdts/libglvnd-devel-cos7-x86_64/libglvnd-license              | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 cdts/libglvnd-devel-cos7-x86_64/meta.yaml                     |  62 ++++++++++++
 cdts/libglvnd-egl-cos7-aarch64/build.sh                       |  59 +++++++++++
 cdts/libglvnd-egl-cos7-aarch64/libglvnd-license               | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 cdts/libglvnd-egl-cos7-aarch64/meta.yaml                      |  50 ++++++++++
 cdts/libglvnd-egl-cos7-ppc64le/build.sh                       |  59 +++++++++++
 cdts/libglvnd-egl-cos7-ppc64le/libglvnd-license               | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 cdts/libglvnd-egl-cos7-ppc64le/meta.yaml                      |  50 ++++++++++
 cdts/libglvnd-egl-cos7-x86_64/build.sh                        |  59 +++++++++++
 cdts/libglvnd-egl-cos7-x86_64/libglvnd-license                | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 cdts/libglvnd-egl-cos7-x86_64/meta.yaml                       |  50 ++++++++++
 cdts/libglvnd-gles-cos7-aarch64/build.sh                      |  59 +++++++++++
 cdts/libglvnd-gles-cos7-aarch64/libglvnd-license              | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 cdts/libglvnd-gles-cos7-aarch64/meta.yaml                     |  50 ++++++++++
 cdts/libglvnd-gles-cos7-ppc64le/build.sh                      |  59 +++++++++++
 cdts/libglvnd-gles-cos7-ppc64le/libglvnd-license              | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 cdts/libglvnd-gles-cos7-ppc64le/meta.yaml                     |  50 ++++++++++
 cdts/libglvnd-gles-cos7-x86_64/build.sh                       |  59 +++++++++++
 cdts/libglvnd-gles-cos7-x86_64/libglvnd-license               | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 cdts/libglvnd-gles-cos7-x86_64/meta.yaml                      |  50 ++++++++++
 cdts/libglvnd-opengl-cos7-aarch64/build.sh                    |  59 +++++++++++
 cdts/libglvnd-opengl-cos7-aarch64/libglvnd-license            | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 cdts/libglvnd-opengl-cos7-aarch64/meta.yaml                   |  47 +++++++++
 cdts/libglvnd-opengl-cos7-ppc64le/build.sh                    |  59 +++++++++++
 cdts/libglvnd-opengl-cos7-ppc64le/libglvnd-license            | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 cdts/libglvnd-opengl-cos7-ppc64le/meta.yaml                   |  47 +++++++++
 cdts/libglvnd-opengl-cos7-x86_64/build.sh                     |  59 +++++++++++
 cdts/libglvnd-opengl-cos7-x86_64/libglvnd-license             | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 cdts/libglvnd-opengl-cos7-x86_64/meta.yaml                    |  47 +++++++++
 legacy_cdts/libglvnd-core-devel-cos7-aarch64/build.sh         |  59 +++++++++++
 legacy_cdts/libglvnd-core-devel-cos7-aarch64/libglvnd-license | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 legacy_cdts/libglvnd-core-devel-cos7-aarch64/meta.yaml        |  44 ++++++++
 legacy_cdts/libglvnd-core-devel-cos7-ppc64le/build.sh         |  59 +++++++++++
 legacy_cdts/libglvnd-core-devel-cos7-ppc64le/libglvnd-license | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 legacy_cdts/libglvnd-core-devel-cos7-ppc64le/meta.yaml        |  44 ++++++++
 legacy_cdts/libglvnd-devel-cos7-aarch64/build.sh              |  66 ++++++++++++
 legacy_cdts/libglvnd-devel-cos7-aarch64/libglvnd-license      | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 legacy_cdts/libglvnd-devel-cos7-aarch64/meta.yaml             |  63 ++++++++++++
 legacy_cdts/libglvnd-devel-cos7-ppc64le/build.sh              |  66 ++++++++++++
 legacy_cdts/libglvnd-devel-cos7-ppc64le/libglvnd-license      | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 legacy_cdts/libglvnd-devel-cos7-ppc64le/meta.yaml             |  63 ++++++++++++
 legacy_cdts/libglvnd-egl-cos7-aarch64/build.sh                |  59 +++++++++++
 legacy_cdts/libglvnd-egl-cos7-aarch64/libglvnd-license        | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 legacy_cdts/libglvnd-egl-cos7-aarch64/meta.yaml               |  50 ++++++++++
 legacy_cdts/libglvnd-egl-cos7-ppc64le/build.sh                |  59 +++++++++++
 legacy_cdts/libglvnd-egl-cos7-ppc64le/libglvnd-license        | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 legacy_cdts/libglvnd-egl-cos7-ppc64le/meta.yaml               |  50 ++++++++++
 legacy_cdts/libglvnd-gles-cos7-aarch64/build.sh               |  59 +++++++++++
 legacy_cdts/libglvnd-gles-cos7-aarch64/libglvnd-license       | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 legacy_cdts/libglvnd-gles-cos7-aarch64/meta.yaml              |  50 ++++++++++
 legacy_cdts/libglvnd-gles-cos7-ppc64le/build.sh               |  59 +++++++++++
 legacy_cdts/libglvnd-gles-cos7-ppc64le/libglvnd-license       | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 legacy_cdts/libglvnd-gles-cos7-ppc64le/meta.yaml              |  50 ++++++++++
 legacy_cdts/libglvnd-opengl-cos7-aarch64/build.sh             |  59 +++++++++++
 legacy_cdts/libglvnd-opengl-cos7-aarch64/libglvnd-license     | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 legacy_cdts/libglvnd-opengl-cos7-aarch64/meta.yaml            |  47 +++++++++
 legacy_cdts/libglvnd-opengl-cos7-ppc64le/build.sh             |  59 +++++++++++
 legacy_cdts/libglvnd-opengl-cos7-ppc64le/libglvnd-license     | 540 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 legacy_cdts/libglvnd-opengl-cos7-ppc64le/meta.yaml            |  47 +++++++++
 77 files changed, 16310 insertions(+)
```

**Comment by basnijholt** [comment]
I discussed with @beckermr on Gitter. This PR should be ready to be merged from the perspective of `libglvnd-devel`. The only "problem" is that I had to manually copy the license files into the folders that were created using the script. I believe he still wants to fix this before we merge this.

**Comment by beckermr** [comment]
Yes. I know you are under pressure at work to get this done. I will try and look today.

**Comment by beckermr** [comment]
Are all the CDTs in this PR licensed under the same license?

**Comment by beckermr** [comment]
We need to check this before we merge. 

When we figure that out, then we need to add yaml entries to the cdt slugs file with these other CDTs, adding their licenses.

File: cdt_slugs.yaml [hunk 1]
```diff
@@ -166,6 +166,18 @@ libglvnd:
...
+libglvnd-devel:
+  custom: false
+  skipped_cdts:
+    - cos6-x86_64
+  license_file: licenses/libglvnd-license
+  build_append:
+    all: |
+      pushd ${SYSROOT_DIR}/usr/lib64 > /dev/null 2>&1
+      ls -lah .
+      popd > /dev/null 2>&1
```
Comment (beckermr) [inline cdt_slugs.yaml:178]: This extra build script doesn't do anything. We should remove it.

**Comment by basnijholt** [comment]
Yes, they are all under the same [license](https://github.com/NVIDIA/libglvnd#libglvnd).

**Comment by basnijholt** [comment]
@beckermr, thanks a lot for working on https://github.com/conda-forge/cdt-builds/pull/45. I will close this PR as it is superseded by that one.

**Pull request closed without merging**